src/importProvider.ts

```diff
@@ -0,0 +1,174 @@
+'use strict';
+import * as vscode from 'vscode';
+import * as Url from 'url';
+import { Uri, ViewColumn } from 'vscode';
+import * as fs from 'fs';
+import * as path from 'path';
+import { guessFileExtension } from './dataUriTextDocumentContentProvider';
+
+export async function load(sourceFilename: string) {
+    if (typeof sourceFilename == 'undefined') {
+        return;
+    }
+    if (!fs.existsSync(sourceFilename)) {
+        return;
+    }
+
+    // Compose an target filename
+    let targetFilename = sourceFilename.replace('.glb', '.gltf');
+
+    if (!vscode.workspace.getConfiguration('glTF').get('neverPromptForFilename')) {
+        const options: vscode.SaveDialogOptions = {
+            defaultUri: Uri.file(targetFilename),
+            filters: {
+                'glTF': ['gltf'],
+                'All files': ['*']
+            }
+        };
+        let uri = await vscode.window.showSaveDialog(options);
+        if (uri !== undefined) {
+            targetFilename = uri.fsPath;
+        }
+        else {
+            return;
+        }
+    }
+
+    // Read the GLB data
+    const Binary = {
+        Magic: 0x46546C67
+    };
+
+    const sourceBuf = fs.readFileSync(sourceFilename);
+    const readMagic = sourceBuf.readUInt32LE(0);
+    if (readMagic !== Binary.Magic) {
+        return;
```

Comment:
Here, and everywhere in this file where you've encountered an error, we should display an error to the user.  Since this is wrapped in try/catch, we can *almost* do this simply by throwing the error from here:

```
        throw new Error('Source file does not appear to be a GLB (glTF Binary) model.');
```

But, this function is `async`, so to catch this error, the call site must await.  I'll leave a separate comment on that.